tests/run-pass/concurrency/libc_prctl_thread_name.rs

```diff
@@ -0,0 +1,18 @@
+// ignore-windows: No libc on Windows
+// ignore-macos: No prctl on MacOS
+
+#![feature(rustc_private)]
+
+extern crate libc;
+
+use std::ffi::CString;
+
+fn main() {
+    unsafe {
+        let thread_name = CString::new("hello").expect("CString::new failed");
+        assert_eq!(libc::prctl(libc::PR_SET_NAME, thread_name.as_ptr() as libc::c_long, 0 as libc::c_long, 0 as libc::c_long, 0 as libc::c_long), 0);
+        let mut buf = [0; 6];
+        assert_eq!(libc::prctl(libc::PR_GET_NAME, buf.as_mut_ptr() as libc::c_long, 0 as libc::c_long, 0 as libc::c_long, 0 as libc::c_long), 0);
+        assert_eq!(thread_name.as_bytes_with_nul(), buf);
+    }
+}
```

Comment:
I think we can probably consolidate some of these run-pass tests into one (e.g., this one here should go into `libc.rs`). I'd like to avoid having too many run-pass tests as we have a quite noticeable constant overhead per test from executing everything before and after the user-supplied `main`.

We can do this is a follow-up PR though if you prefer (in that case, let's open an issue so we do not forget).